### DISCUSSION TITLE
Allow admin approval of pending invites

### DISCUSSION THREAD

**Discussion by client4** [docmost/docmost#37] [Help / Questions] [answered]
I don't have many people to add to my Docmost instance and don't want to take the time to set up SNMP.  It would be great to approve pending-invite members from the settings panel for admins, and to either set a temporary password for them or send them a magic link.

**Comment by Philipinho** [comment]
For now, you can copy the invitation id and token from the "workspace_invitations" table.
Then goto `https://yoursite.com/invites/INVITATION_ID_HERE?token=INVITATION_TOKEN_HERE`. This is where the invited user signs up.

**Comment by client4** [comment]
Thanks for the workaround!

**Reply by client4** [reply]
For those who find this,
* Connect to the  db docker `docker exec -it CONTAINER_NAME bash`
* Connect into the db `psql -U docmost`
* See all entries in the invitation table `select * from workspace_invitations;`
* Then create a link for the user to sign in.  Note that if you're creating the user for someone else an incognito window will need to be used. `https://yoursite.com/invites/INVITATION_ID_HERE?token=INVITATION_TOKEN_HERE`

For example, a URL you could share would look like `https://yoursite.com/invites/01907078-a64b-734c-86f1-ae1fc777bfb0?token=9j1tonjhv055lzez`

**Reply by H4rmey** [reply]
To add to this: 
For now i've created a quick and dirty script that will generate the link.

```bash

if [[ $# -ne 2 ]]; then
  echo "Usage   : $0 <your site> <username>"
  echo "Example : $0 https://yoursite.com foo@bar.com"
  exit
fi

token=$(docker exec -it docmost-db-1 psql -U docmost -c 'select email,token from workspace_invitations;' | grep ${2} | awk -F'|' '{print $2}' | sed 's/ //g;s/\r//g;s/\$//g')
id_key=$(docker exec -it docmost-db-1 psql -U docmost -c 'select email,id from workspace_invitations;' | grep ${2} | awk -F'|' '{print $2}' | sed 's/ //g;s/\r//g;s/\$//g')

[[ -z ${token} ]] && echo "Error token not generated" && exit
[[ -z ${id_key} ]] && echo "Error id_key not generated" && exit

echo "$1/invites/${id_key}?token=${token}"
```

Don't just copy and paste! this is just a quick script that works for me


This works by first creating an invite in docmost and then running this script on your server to generate the link by getting the id/token from the database

**Comment by bo3bdo** [comment]
i hop admin approval soon

**Comment by FromNetToDev** [comment]
Hello everyone. Why this recommendation is not work. When access to link with browser https://yoursite.com/invites/INVITATION_ID_HERE?token=INVITATION_TOKEN_HERE I got "invalid invitation link"?

**Reply by FromNetToDev** [reply]
@Philipinho

**Reply by Philipinho** [reply]
Please check your server logs for any errors.

**Reply by FromNetToDev** [reply]
The fact of the matter is that there is nothing new in the logs for the last 8 hours. and during this time I probably tried to add a user 100 times. I remind you that there are no logs in either the postgres container or the docmost container

**Reply by FromNetToDev** [reply]
in the browser in developer mode (f12) in the network tab, when requesting I receive info with the following content ({"message":"Invitation not found","error": "Not found","statusCode":404})

**Reply by FromNetToDev** [reply]
POSThttp://ip_addres:3000/api/workspace/invites/info 404 (Not Found)

**Reply by FromNetToDev** [reply]
but the api work, curl http://ip_address:3000/api 
{"data":"Hello World!","success":true,"status":200}

**Reply by Philipinho** [reply]
Can you find the invitation under the pending invites list?

**Reply by FromNetToDev** [reply]
yes

**Reply by FromNetToDev** [reply]
in this section: Settings->Workspace->Members->Pending

**Reply by ja49619** [reply]
try logout before or incognito mode

**Reply by AnatoliSavitskiAll** [reply]
@FromNetToDev  INVITATION_ID_HERE. it is Not an invited_by_id. Use the column "id" here.

**Comment by SyndicatedPillbug** [comment]
I heartily second this feature request. Manually adding users is essential for my use-case and smtp isn't something I need or want to deal with for this.

**Reply by JonathanDHolden** [reply]
ooh... that would be lovely!

**Comment by securenotebook** [comment]
Hi All, 

i dont have a SMTP server nor do i want to create one, i built on @H4rmey example to create some scripts that allow you to list and also create invites from a CSV file. 

https://github.com/securenotebook/docmostInvite - Some others may find this useful, only spent a couple of hours on this so its a little rough around the edges!

**Comment by Philipinho** [comment, marked as answer]
Admins can now copy user invite links from v0.8.4

Release: https://github.com/docmost/docmost/releases/tag/v0.8.4

**Reply by securenotebook** [reply]
Oh thank you so much, I completly missed that!  - Its really a great product - I feel a little stupid now, but i learned something new! 

listing is now a little useless, but some people may find creating from a CSV file useful. Unless ive missed this too?

**Reply by Philipinho** [reply]
Might be redundant since you can already paste a list of space or comma separated email addresses to be invited.

**Reply by securenotebook** [reply]
might == 100% ! 
thanks again, youre right its not needed!  Oh well, i'll leave it as a reminded for people to do more research before they run off building something.

**Reply by mupeifeiyi** [reply]
Hello, I am using version 0.8.4. After inviting members, I saw the copy link button, and the response after clicking it was also copied successfully. But in fact, there is no inviteLink in the clipboard. I got it from the response of the http://xxx/api/workspace/invites/link interface.
<img width="1199" alt="1741747405009" src="https://github.com/user-attachments/assets/7b93fde9-d450-413b-aaab-318ae2d49e3b" />

**Reply by Philipinho** [reply]
@mupeifeiyi make sure you have https enabled.

**Reply by mupeifeiyi** [reply]
> @mupeifeiyi确保您已启用 https。
Does this part require mandatory https? I just found that after creating a page, I can write the title, but I can't write the body content.

**Reply by mupeifeiyi** [reply]
> @mupeifeiyi make sure you have https enabled.
I have enabled https, and the copy link works fine. I still hope that the later versions can also support http. In addition, I cannot write the body when creating a page. Do you have any suggestions?

**Reply by Ringoinmel** [reply]
same problem.  click copy link but got nothing

**Reply by Ringoinmel** [reply]
Click copy link in pending section and then paste it to web browser. But I need to do it in ubuntu web browser(it doesn't work in windows environment). Also, make sure the member is logout.
